assets/js/components/settings/SettingsAdmin.js

```diff
@@ -35,15 +35,87 @@ import ResetButton from '../ResetButton';
 import SettingsCardKeyMetrics from './SettingsCardKeyMetrics';
 import SettingsPlugin from './SettingsPlugin';
 import ConnectedIcon from '../../../svg/icons/connected.svg';
+import PreviewBlock from '../PreviewBlock';
 const { useSelect } = Data;
 
 export default function SettingsAdmin() {
+	const isAnalyticsConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
+	);
 	const showKeyMetricsSettings = useSelect(
 		( select ) =>
-			select( CORE_MODULES ).isModuleConnected( 'analytics-4' ) &&
+			isAnalyticsConnected &&
 			select( MODULES_SEARCH_CONSOLE ).isGatheringData() === false &&
 			select( MODULES_ANALYTICS_4 ).isGatheringData() === false
 	);
+	const showKeyMetricsSettingsLoading = useSelect( ( select ) => {
+		if (
+			! select( CORE_MODULES ).hasFinishedResolution(
+				'isModuleConnected',
+				[ 'analytics-4' ]
+			)
+		) {
+			return true;
+		}
+
+		// Following resolvers are not resolved if Analytics is disconnected.
```

Comment:
```suggestion
		// The resolvers below are never resolved if Analytics is disconnected,
		// so if it's disconnected, return early.
		//
		// Because they're never called nothing else can be loading.
```